src/test/java/org/opentripplanner/apis/gtfs/mapping/routerequest/RouteRequestMapperScooterTest.java

```diff
@@ -0,0 +1,204 @@
+package org.opentripplanner.apis.gtfs.mapping.routerequest;
+
+import static java.util.Map.entry;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.opentripplanner.apis.gtfs.mapping.routerequest.RouteRequestMapperTest.createArgsCopy;
+import static org.opentripplanner.apis.gtfs.mapping.routerequest.RouteRequestMapperTest.executionContext;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.framework.model.Cost;
+import org.opentripplanner.routing.core.VehicleRoutingOptimizeType;
+
+class RouteRequestMapperScooterTest {
+
+  @Test
+  void testBasicScooterPreferences() {
+    var scooterArgs = createArgsCopy(RouteRequestMapperTest.ARGS);
+    var reluctance = 7.5;
+    var speed = 15d;
+    scooterArgs.put(
+      "preferences",
+      Map.ofEntries(
+        entry(
+          "street",
+          Map.ofEntries(
+            entry("scooter", Map.ofEntries(entry("reluctance", reluctance), entry("speed", speed)))
+          )
+        )
+      )
+    );
+    var env = executionContext(scooterArgs, Locale.ENGLISH, RouteRequestMapperTest.CONTEXT);
+    var routeRequest = RouteRequestMapper.toRouteRequest(env, RouteRequestMapperTest.CONTEXT);
+    var scooterPreferences = routeRequest.preferences().scooter();
+    assertEquals(reluctance, scooterPreferences.reluctance());
+    assertEquals(speed, scooterPreferences.speed());
+  }
+
+  @Test
+  void testScooterTrianglePreferences() {
+    var scooterArgs = createArgsCopy(RouteRequestMapperTest.ARGS);
+    var scooterSafety = 0.3;
+    var scooterFlatness = 0.5;
+    var scooterTime = 0.2;
+    scooterArgs.put(
+      "preferences",
+      Map.ofEntries(
+        entry(
+          "street",
+          Map.ofEntries(
+            entry(
+              "scooter",
+              Map.ofEntries(
+                entry(
+                  "optimization",
+                  Map.ofEntries(
+                    entry(
+                      "triangle",
+                      Map.ofEntries(
+                        entry("safety", scooterSafety),
+                        entry("flatness", scooterFlatness),
+                        entry("time", scooterTime)
+                      )
+                    )
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
```

Comment:
The utility methods and test is in the same class because it was easy for me to do, and post - I would splitt it. If you put the utility class in the test code or main does not matter much.

Reply:
Otherwise I like this idea, but the problem is that we can't really parse into custom objects such as OffsetDateTime from a string in the json as the `toMap` is not aware of the schema/scalars. Perhaps it would be possible to use the graphql code to parse json into a map or directly into a `DataFetchingEnvironment`. Do you want me to investigate that possibility?

Reply:
Ok, not know. But, we should find a solution for this. But, we could take one step back and look at the best way to automate and test the API.